packages/lmi/src/lmi/llms.py

```diff
@@ -67,6 +69,52 @@
     "refusal",  # Anthropic safety filter
 )
 
+
+def parse_cached_usage(usage) -> tuple[int, int]:
+    """Parse cached token counts from LiteLLM usage object.
+
+    Args:
+        usage: LiteLLM usage object (from ModelResponse.usage or streaming chunk.usage)
+
+    Returns:
+        Tuple of (cache_read_tokens, cache_creation_tokens)
+
+    Note:
+        - OpenAI: Uses prompt_tokens_details.cached_tokens (read only)
+        - Anthropic: Uses cache_read_input_tokens and cache_creation_input_tokens
+        - Returns (0, 0) if no cached tokens found or usage is None
+    """
+    cache_read = 0
+    cache_creation = 0
+
+    if not usage:
+        return cache_read, cache_creation
+
+    # Cache reads: Try normalized format first, then Anthropic-specific
+    # LiteLLM *usually* normalizes to prompt_tokens_details.cached_tokens,
+    # but Anthropic responses may have cache_read_input_tokens at top level
+    if hasattr(usage, "prompt_tokens_details"):
+        prompt_details = getattr(usage, "prompt_tokens_details", None)
+        if prompt_details:
+            cached_val = (
+                prompt_details.get("cached_tokens", 0)
+                if isinstance(prompt_details, dict)
+                else getattr(prompt_details, "cached_tokens", 0)
```

Comment:
Let's not inject these `0` defaults -- in general we do not want to inject information